### DISCUSSION TITLE
Scrubbing/Volume censoring with Halfpipe

### DISCUSSION THREAD

**Discussion by Ammar-Elsaeed** [HALFpipe/HALFpipe#745] [Q&A] [answered]
Hello All,
I am using the latest Halfpipe image with singularity to preprocess rs-fMRI data. Particularly, I want to do standard preprocessing steps with a focus on mitigating motion artifact, and ultimately use an atlas to get atlas-based connectivity, thankfully, all steps possible to implement within the Amazing Gui. However, I also want to do "Volume Censoring" (Also know as scrubbing), this can not be specified in the GUI, but I found some parameters in the generated spec.json that might relate to that, so wanted to ask if Volume censoring is possible with Halfpipe, and how to do it in this case.

Additionaly, I am new to this, so wanted to ask How does halfpipe do Volume censoring? does Halfpipe only output a mask that indicates which volumes should be censored from the ROI timeseries (after parcellation), or does it use that info at an intermediate step?

In other words, if I run atlas-based connectivity with Halfpipe without setting volume censoring, can I use the output Frame-wise displacement (FD) values  and the ROI timeseries for subject X to do volume censoring on the ROI timeseries by a custom script/manually then, or is it a step that should be specified and done by Halfpipe during preprocessing?

FInally, here is a snippet of the spec.json in case it is relevant for doing volume censoring:
```
"halfpipe_version": "1.2.2.post1.dev749+g34cf879",
    "schema_version": "3.0",
    "timestamp": "2024-08-29_12-34",
    "global_settings": {
        "dummy_scans": 0,
        "slice_timing": true,
        "use_bbr": null,
        "skull_strip_algorithm": "ants",
        "run_mriqc": false,
        "run_fmriprep": true,
        "run_halfpipe": true,
        "fd_thres": 0.5,
        "anat_only": false,
        "write_graph": false,
        "hires": false,
        "run_reconall": false,
        "t2s_coreg": false,
        "medial_surface_nan": false,
        "bold2t1w_dof": 9,
        "fmap_bspline": true,
        "force_syn": false,
        "longitudinal": false,
        "regressors_all_comps": false,
        "regressors_dvars_th": 1.5,
        "regressors_fd_th": 0.5,
        "skull_strip_fixed_seed": false,
        "skull_strip_template": "OASIS30ANTs",
        "run_aroma": true,
        "aroma_err_on_warn": false,
        "aroma_melodic_dim": -200,
        "sloppy": false
    },

``` 

Thanks a lot for your help and for this great tool, and Kind regards!

**Comment by HippocampusGirl** [comment, marked as answer]
Hi @Ammar-Elsaeed,

I'm sorry, we currently do not support scrubbing or censoring, but it's high on our list to support!
